bloxid/v0.go

```diff
@@ -48,16 +50,78 @@ var (
 
 // NewV0 parse a string into a typed guid, return an error
 // if the string fails validation.
+/*
 func NewV0(bloxid string) (*V0, error) {
 	return parseV0(bloxid)
 }
+*/
 
 const (
 	V0Delimiter   = "."
 	bloxidTypeLen = 4
 )
 
-func parseV0(bloxid string) (*V0, error) {
+type EncodeDecodeOpts func(o *V0Options)
+type NewV0Opts func(fnOpts ...EncodeDecodeOpts) *V0Options
+
+// NewV0 parse a string into a typed guid, return an error
+// if the string fails validation.
+
+func NewV0(bloxid string, opts ...*V0Options) (*V0, error) {
+	var inOpts *V0Options
+
+	if len(opts) > 1 {
```

Comment:
no need to check len opts or create opts, parse them here - and pass the single V0Options to the GenerateV0() and the parseV0() funcs